frame/support/src/traits/tokens/fungible/balanced.rs

```diff
@@ -0,0 +1,359 @@
+	fn set_balance(who: &AccountId, amount: Self::Balance) -> DispatchResult;
+
+	/// Set the total issuance to `amount`.
+	fn set_total_issuance(amount: Self::Balance);
+
+	/// Reduce the balance of `who` by `amount`. If it cannot be reduced by that amount for
+	/// some reason, return `Err` and don't reduce it at all. If Ok, return the imbalance.
+	///
+	/// Minimum balance will be respected and the returned imbalance may be up to
+	/// `Self::minimum_balance() - 1` greater than `amount`.
+	fn decrease_balance(who: &AccountId, amount: Self::Balance)
+		-> Result<Self::Balance, DispatchError>
+	{
+		let old_balance = Self::balance(who);
+		let (mut new_balance, mut amount) = if old_balance < amount {
+			Err(TokenError::NoFunds)?
+		} else {
+			(old_balance - amount, amount)
+		};
+		if new_balance < Self::minimum_balance() {
+			amount = amount.saturating_add(new_balance);
+			new_balance = Zero::zero();
+		}
+		// Defensive only - this should not fail now.
+		Self::set_balance(who, new_balance)?;
+		Ok(amount)
+	}
+
+	/// Reduce the balance of `who` by the most that is possible, up to `amount`.
+	///
+	/// Minimum balance will be respected and the returned imbalance may be up to
+	/// `Self::minimum_balance() - 1` greater than `amount`.
+	///
+	/// Return the imbalance by which the account was reduced.
+	fn decrease_balance_at_most(who: &AccountId, amount: Self::Balance)
+		-> Self::Balance
+	{
+		let old_balance = Self::balance(who);
+		let (mut new_balance, mut amount) = if old_balance < amount {
+			(Zero::zero(), old_balance)
+		} else {
+			(old_balance - amount, amount)
+		};
+		let minimum_balance = Self::minimum_balance();
+		if new_balance < minimum_balance {
+			amount = amount.saturating_add(new_balance);
+			new_balance = Zero::zero();
+		}
+		let mut r = Self::set_balance(who, new_balance);
+		if r.is_err() {
+			// Some error, probably because we tried to destroy an account which cannot be destroyed.
+			if amount > minimum_balance {
+				new_balance += minimum_balance;
+				amount -= minimum_balance;
+				r = Self::set_balance(who, new_balance);
+			}
+			if r.is_err() {
+				// Still an error. Apparently it's not possibl to reduce at all.
+				amount = Zero::zero();
+			}
+		}
+		amount
+	}
+
+	/// Increase the balance of `who` by `amount`. If it cannot be increased by that amount
+	/// for some reason, return `Err` and don't increase it at all. If Ok, return the imbalance.
+	///
+	/// Minimum balance will be respected and an error will be returned if
+	/// `amount < Self::minimum_balance()` when the account of `who` is zero.
+	fn increase_balance(who: &AccountId, amount: Self::Balance)
+		-> Result<Self::Balance, DispatchError>
+	{
+		let old_balance = Self::balance(who);
+		let new_balance = old_balance.saturating_add(amount);
+		if new_balance < Self::minimum_balance() {
+			Err(TokenError::BelowMinimum)?
+		}
+		if old_balance != new_balance {
+			Self::set_balance(who, new_balance)?;
+		}
+		Ok(amount)
```

Comment:
amount here could also be inaccurate due to saturation above. (although i recommend we just used checked)
```suggestion
		Ok(new_balance - old_balance)
```